dfd-wrapper/src/main/java/com/revolut/rxdata/dfd_wrapper/DataFlowDelegate.kt

```diff
@@ -0,0 +1,109 @@
+package com.revolut.rxdata.dfd_wrapper
+
+import com.revolut.data.model.Data
+import com.revolut.rxdata.dod.DataObservableDelegate
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.rx2.asFlow
+import kotlinx.coroutines.rx2.await
+import kotlinx.coroutines.rx2.rxSingle
+
+/*
+ * Copyright (C) 2023 Revolut
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+class DataFlowDelegate<Params : Any, Domain : Any>(
+    fromNetwork: suspend (params: Params) -> Domain,
+    fromMemory: (params: Params) -> Domain?,
+    toMemory: (params: Params, Domain) -> Unit,
+    fromStorage: suspend (params: Params) -> Domain?,
+    toStorage: suspend (params: Params, Domain) -> Unit,
+    onRemove: suspend (params: Params) -> Unit = { _ -> Unit },
+) {
+    private val inner = DataObservableDelegate<Params, Domain>(
+        fromNetwork = { params ->
+            rxSingle(DataFlowDelegateDispatchers.ioDispatcher()) {
```

Comment:
`DataObservableDelegate` switches thread to `IO` internally, so this thread switching is not needed here. We can use `Unconfined` dispatcher instead

Reply:
This was mostly done for testing, e.g. to substitute `UnconfinedTestDispatcher` which is better integrated with testing infrastructure. I tried to hardcode `Dispatchers.Unconfined`, and 6 tests failed in `DataFlowDelegateTest`. I'd vote to keep `DataFlowDelegateDispatchers`, but use `Unconfined` by default instead of `IO`.

Reply:
Got it. Keeping `DataFlowDelegateDispatchers` and using `Unconfined` sounds good to me 👍

Reply:
Updated the PR. Kept `IO` to wrap `update*` methods.